packages/editor/src/components/post-trash/index.js

```diff
@@ -20,21 +24,40 @@ export default function PostTrash() {
 		};
 	}, [] );
 	const { trashPost } = useDispatch( editorStore );
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
 
 	if ( isNew || ! postId ) {
 		return null;
 	}
 
+	const handleConfirm = () => {
+		setShowConfirmDialog( false );
+		trashPost();
+	};
+
 	return (
-		<Button
-			className="editor-post-trash"
-			isDestructive
-			variant="secondary"
-			isBusy={ isDeleting }
-			aria-disabled={ isDeleting }
-			onClick={ isDeleting ? undefined : () => trashPost() }
-		>
-			{ __( 'Move to trash' ) }
-		</Button>
+		<>
+			<Button
+				className="editor-post-trash"
+				isDestructive
+				variant="secondary"
+				isBusy={ isDeleting }
+				aria-disabled={ isDeleting }
+				onClick={
+					isDeleting ? undefined : () => setShowConfirmDialog( true )
+				}
+			>
+				{ __( 'Move to trash' ) }
+			</Button>
+			<ConfirmDialog
+				isOpen={ showConfirmDialog }
+				onConfirm={ handleConfirm }
+				onCancel={ () => setShowConfirmDialog( false ) }
+			>
+				{ __(
+					'Are you sure you want to move this post to the trash?'
```

Comment:
We should probably avoid the term "post" or contextualize with whatever post type is the current context.

Reply:
I agree. Current context would be best here.

Reply:
Should we consider this as a blocker? cc @jasmussen

Reply:
I agree current context would be best, but since we are currently shipping in trunk _without_ the warning, it seems an improvement to _add_ one as a first step, but we can just rephrase to this:

> Are you sure you want to move this to the trash?

Reply:
I just checked, and "switch to draft" also uses the "post" term in the confirmation message. I think we can merge this as it is and contextualize both in follow-up PR.